### ISSUE TITLE
 Strange water with ArchaicFix

### ISSUE THREAD

**Issue by kotmatross28729** [embeddedt/ArchaicFix#57]
I don’t know if there is already such an error here or not, but I didn’t seem to find it (or maybe I didn’t search well)

### **note**:  manipulations with `disableBlockingChunkUpdates` and `enableThreadedChunkUpdates` (turn on, off) do not help

**Comment by makamys** [comment]
Does the 1st issue only happen with OptiFine?

The 2nd one sounds like a duplicate of #19.

**Comment by kotmatross28729** [comment]
no, it's the same with dynamic light

**Comment by kotmatross28729** [comment]
I was able to reproduce with a TNT block, apparently, this happens very rarely with all blocks, but always with liquids. (this does not happen if dynamic lighting is disabled)

https://user-images.githubusercontent.com/110309314/208972638-d5336de3-74ed-46b8-8b34-844beb237d6b.mp4

**Comment by makamys** [comment]
A user on Discord also reported what appears to be the same issue:

> Not sure if AF is the cause or if its my GPU hating AF, but if Dynlights is active at all while moving, weird flickering of chunks likes to happen. Usually happens in areas that have had any types of edits in them. (blocks being broken or placed) 

https://cdn.discordapp.com/attachments/993717703032447016/1050943346773868634/javaw_kfahbwNq7Y_1.mp4

I was able to reproduce the issue (using water) with everything in AF's config disabled, so the incompatibility is with one of the mandatory tweaks.

**Comment by kotmatross28729** [comment]
Is this fixable or do It need to completely rewrite the code of something?
This is easy to  reproduce with blocks, if you break them very often and immediately place them in, as I understand it, any liquids are updated very often, and this flickering is connected precisely with updating blocks

**Comment by makamys** [comment]
I compiled a version of the mod that has the lighting engine disabled (that is, all mixins in the package `archaicfix.mixins.*.lighting`), and the issue went away. So the problem is in that part of the mod. That's all I can say.

**Comment by embeddedt** [comment]
I am pretty sure it's not an issue with the standalone Dynamic Lights mod. Not sure if that works alongside OptiFine.

**Comment by MagmaBro123** [comment]
It does iirc, they have an override option in.. 1.12 I think.

**Comment by makamys** [comment]
> I am pretty sure it's not an issue with the standalone Dynamic Lights mod

It is, it's what I tested with.

We could always add an option to disable the lighting engine part of the mod if this proves too difficult to fix properly.

**Comment by embeddedt** [comment]
> We could always add an option to disable the lighting engine part of the mod if this proves too difficult to fix properly.

This has been added and will be in the next release.

**Comment by calreveraster** [comment]
Just throwing out there that as of version 0.5.0 this is still happening, even with the lighting engine disabled in config - but as far as I can tell, it only happens with water. It is worse with dynamic lights from OptiFine enabled, but will still (extremely rarely) happen without it. 

The easiest way to reproduce the issue (without dynamic lights) is to spill water, then break the source block - then spill water in the same spot a second time, and youll see it occur in spaces that the water will be occupying once it's completely propagated.

I've tested it with every option in the config disabled. 

It also does seem to occasionally happen to slopes from carpenter's blocks, if that helps. 

If needed I'll see if I can grab some videos a bit later - at work at the moment so I can't really sit down and capture anything.

**Comment by calreveraster** [comment]
Another update: 

hodgepodge, tentatively, fixes the issue. I assume it's the openGL state error fixes that they have in the mod. I'd encourage anyone with the issue to drop hodgepodge into the pack and see if it persists, or if it's just me.

**Comment by kotmatross28729** [comment]
> Another update: hodgepodge, tentatively, fixes the issue. I assume it's the openGL state error fixes that they have in the mod. I'd encourage anyone with the issue to drop hodgepodge into the pack and see if it persists, or if it's just me.

I have hodgepodge in my pack but still have this error

**Comment by calreveraster** [comment]
> > Another update: hodgepodge, tentatively, fixes the issue. I assume it's the openGL state error fixes that they have in the mod. I'd encourage anyone with the issue to drop hodgepodge into the pack and see if it persists, or if it's just me.
> 
> I have hodgepodge in my pack but still have this error

What other mods are you using that impact rendering? 

my current layout is optifine(with shaders)+archaicfix+hodgepodge+falsetweaks

I used fastcraft for years, but with the updates to these mods (and it's issues with shaders) I've found I no longer need it

**Comment by kotmatross28729** [comment]
Exactly the same, + FlyingCullers (not sure, but maybe CoreTweaks and bugtorch do something too)

**Comment by MagmaBro123** [comment]
AF on its own, with Dynamic Lights, (besides UniMixins) both with no config changes causes this issue.

**Comment by calreveraster** [comment]
Just a small update - still happening on 0.6.1 - and it appears Magma was correct, hodgepodge is not fixing the problem. It just seems to be very intermittent for me, and for a while I didnt notice it, but I'm noticing it again. Sorry for the false positive on that one.

**Comment by makamys** [comment]
It's known that the MC-129 fix causes flickering issues (which is why I [added the flag to disable it](https://github.com/embeddedt/ArchaicFix/commit/2c3a60e39c85238e2f8f8fffbfba7846e2541bb1)), but afaik it's a separate issue from the dynamic lights one, they just have extremely similar symptoms.

MC-129 refers to [this issue](https://bugs.mojang.com/browse/MC-129) on Mojira, and the fix for it was implemented following [this comment](https://bugs.mojang.com/browse/MC-129?focusedId=46972&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-46972).

**Comment by kotmatross28729** [comment]
Well, if the MC-129 fix is ​​known to cause the most significant flickering, might it be worth adding it to the config instead of using System.getProperty? Also, it seems that the problem with dynamic lights is not related to ArchaicFix, as it can be reproduced without it (ArchaicFix).
 
 I can also say that they have slightly different behavior:
* The problem with the MC-129 fix is ​​most noticeable, as the flickering blocks remain until block updating is stopped, or the chunk is reloaded with F3+A. 

* The problem with the dynamic lights itselves is less noticeable, since it don't always appear, and this flickering seems to be a one-time flicker (appears only once and then disappears), and it's visible for about 1 tick.

**Comment by MerrittK** [comment]
How do you use the JVM flag?

**Comment by kotmatross28729** [comment]
> How do you use the JVM flag?

You can add archaicFix.disableMC129 to the environment variables with the true value , 
 or you can add `-DarchaicFix.disableMC129=true` to the java arguments in your launcher
